packages/swarm-components/src/Dropdown.jsx

```diff
@@ -0,0 +1,462 @@
+/* eslint-disable */
+import React, {
+	createContext,
+	Children,
+	useContext,
+	useState,
+	useEffect,
+	useRef,
+} from 'react';
+import Portal from '@reach/portal';
+import Rect, { useRect } from '@reach/rect';
+import WindowSize from '@reach/window-size';
+import { node, func, object, string, number, oneOfType, any } from 'prop-types';
+import { wrapEvent, checkStyles, assignRef } from '@reach/utils';
+import { ForwardedButton } from './Button';
+
+// TODO: add the mousedown/drag/mouseup to select of native menus, will
+// also help w/ remove the menu button tooltip hide-flash.
+
+// TODO: add type-to-highlight like native menus
+
+let MenuContext = createContext();
+
+let checkIfAppManagedFocus = ({ refs, state }) => {
+	if (!state.isOpen && !!refs.menu) {
+		return !refs.menu.contains(document.activeElement);
+	}
+	return false;
```

Comment:
There is a fair amount of 'negative' logic in this function that would be nice to turn around so that it's easier to read. In my head, it currently reads as "If the thing is not open and not-not a menu, then focus is managed if the menu does not contain the active element", which is a bit of a mind bender.

consider

```js
const checkIfAppManagedFocus = ({ refs, state }) =>
  state.isOpen && Boolean(refs.menu) && refs.menu.contains(document.activeElement);
```

which reads as "the focus is managed if and only if the thing is open and has a menu containing the active element"